larrecodnn/ImageMaker/SavePiMu_tool.cc

```diff
@@ -0,0 +1,150 @@
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "art/Utilities/ToolMacros.h"
+#include "canvas/Persistency/Common/FindManyP.h"
+
+#include "larcore/Geometry/Geometry.h"
+#include "lardataobj/RecoBase/Hit.h"
+#include "lardataobj/RecoBase/Track.h"
+#include "lardataobj/RecoBase/TrackHitMeta.h"
+#include "lardataobj/RecoBase/Wire.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+
+#include "ImageMaker.h"
+#include "TTimeStamp.h"
+#include "hep_hpc/hdf5/Ntuple.hpp"
+#include "hep_hpc/hdf5/make_ntuple.hpp"
+
+using namespace hep_hpc::hdf5;
+
+namespace dnn {
+
+  void saveImage(art::Event const& e, hep_hpc::hdf5::File& hdffile)
+  {
+
+    art::ServiceHandle<geo::Geometry> geom;
+
+    static Ntuple<unsigned int, unsigned int, unsigned int, double> evtids(
+      hdffile, "evtids", {"run", "subrun", "event", "evttime"});
+
+    static auto image =
+      make_ntuple({hdffile, "image", 1000},
+                  make_scalar_column<unsigned short>("label"),
+                  make_column<float, 2>(
+                    "adc",                                // 2 means each element is a 2-d array
+                    {50, 50},                             // extent of each array dimension
+                    1024 * 1024 / (2500 * sizeof(float)), // chunk size
+                    {PropertyList{H5P_DATASET_CREATE}(&H5Pset_shuffle)(&H5Pset_deflate, 6u)}));
```

Comment:
Generally, it is better to pass in the `evtids` and `image` objects as arguments to `saveImage(...)` instead of having function-local static variables.  Normally, I'd suggest you make that change, but I'm worried about what the interface would look like with all of the template parameters in the argument list.

I think in this case, we can probably leave things as you left them.

Reply:
I agree it is probably good to keep things as they are to make the tool interface most general.